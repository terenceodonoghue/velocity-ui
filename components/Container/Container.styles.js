import styled from '@emotion/styled';

export const Card = styled.div`
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46, 91, 255, 0.08);
  box-shadow: 0 10px 20px 0 rgba(46, 91, 255, 0.07);
  background-color: var(--color-white);
`;

export const Heading = styled.h3`
  margin: 0 0 26px 0;
  color: var(--color-bluey-grey);
  font-size: 13px;
  font-weight: var(--font-weight-normal);
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
