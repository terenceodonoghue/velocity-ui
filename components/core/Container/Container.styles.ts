import { css } from '@emotion/react';

export const container = css`
  position: relative;
  max-width: 1360px;
  padding: 64px 110px;
`;

export const heading = css`
  margin: 0 0 24px;
`;
