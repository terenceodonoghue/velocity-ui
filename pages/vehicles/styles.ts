import { css, FunctionInterpolation, Theme } from '@emotion/react';

export const filter = css`
  padding: 32px 24px 3px;
`;

export const sliderInput = css`
  margin: 0 6px 34px;
`;

export const textInput: FunctionInterpolation<Theme> = (theme) => css`
  margin: 0 6px 19px;
  color: ${theme.colors.black};
  font-size: 15px;
  font-weight: ${theme.typography.fontWeightRegular};
  letter-spacing: normal;
  line-height: 1.47;
  text-transform: none;

  input {
    color: ${theme.colors.blue};
  }
`;

export const vehicles = css`
  flex: 4;
`;
