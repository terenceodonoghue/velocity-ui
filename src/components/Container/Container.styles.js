import styled from '@emotion/styled';

export const Card = styled.div`
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46, 91, 255, 0.08);
  box-shadow: 0 10px 20px 0 rgba(46, 91, 255, 0.07);
  padding: 32px 24px;
  background-color: ${(props) => props.theme.colors.white};
`;

export const Heading = styled.h3`
  margin: 0 0 26px 0;
  color: ${(props) => props.theme.colors.blueyGrey};
  font-size: 13px;
  font-weight: ${(props) => props.theme.typography.fontWeightRegular};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
