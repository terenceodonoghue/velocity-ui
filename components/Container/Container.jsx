/** @jsx jsx */
import { jsx } from '@emotion/core';
import PropTypes from 'prop-types';
import * as S from './Container.styles';

const Card = ({ children, title, ...rest }) => (
  <S.Card {...rest}>
    {title && <S.Title>{title}</S.Title>}
    {children}
  </S.Card>
);

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  title: PropTypes.string,
};

Card.defaultProps = {
  title: null,
};

export default {
  Card,
};
