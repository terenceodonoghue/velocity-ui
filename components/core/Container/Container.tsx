import React, { FunctionComponent } from 'react';
import * as css from './Container.styles';
import type { ContainerProps } from './Container.types';

const Container: FunctionComponent<ContainerProps> = ({
  children,
  heading,
  ...props
}) => (
  <>
    {heading && (
      <h1 css={css.heading} data-testid="container-heading">
        {heading}
      </h1>
    )}
    <div css={css.container} data-testid="container" {...props}>
      {children}
    </div>
  </>
);

export default Container;
