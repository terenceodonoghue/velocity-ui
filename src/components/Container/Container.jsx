/** @jsx jsx */
import { jsx } from '@emotion/core';
import React from 'react';
import PropTypes from 'prop-types';
import * as S from './Container.styles';

const Card = React.forwardRef(({ children, heading, ...rest }, ref) => (
  <S.Card ref={ref} {...rest}>
    {heading && <S.Heading>{heading}</S.Heading>}
    {children}
  </S.Card>
));

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  heading: PropTypes.string,
};

Card.defaultProps = {
  heading: null,
};


export default {
  Card,
};
