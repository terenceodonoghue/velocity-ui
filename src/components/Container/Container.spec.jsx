import React from 'react';
import { shallow } from 'enzyme';
import faker from 'faker';
import Container from './Container';
import * as S from './Container.styles';

describe('Container', () => {
  describe('Card', () => {
    const heading = faker.lorem.word();
    const words = faker.lorem.words();

    it('should render a heading when present', () => {
      const component = shallow(
        <Container.Card heading={heading}><p>{words}</p></Container.Card>,
      );

      expect(
        component.contains(<S.Heading>{heading}</S.Heading>),
      ).toBe(true);
    });

    it('should not render a heading when absent', () => {
      const component = shallow(
        <Container.Card><p>{words}</p></Container.Card>,
      );

      expect(
        component.contains(<S.Heading>{heading}</S.Heading>),
      ).toBe(false);
    });

    it('should render a child component', () => {
      const component = shallow(
        <Container.Card><p>{words}</p></Container.Card>,
      );

      expect(
        component.contains(<p>{words}</p>),
      ).toBe(true);
    });
  });
});
