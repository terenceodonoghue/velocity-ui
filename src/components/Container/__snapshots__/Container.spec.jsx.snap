// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Card matches snapshot 1`] = `
.emotion-2 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: var(--color-white);
}

.emotion-0 {
  margin: 0 0 26px 0;
  color: var(--color-bluey-grey);
  font-size: 13px;
  font-weight: var(--font-weight-normal);
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

<div
  className="emotion-2 emotion-3"
>
  <h3
    className="emotion-0 emotion-1"
  >
    Card Heading
  </h3>
  Card Content
</div>
`;
